### PR TITLE
fix for issue #178

### PR DESCRIPTION
**Pull request by yannpaul**
This is a fix for issue #178 

Reordered building of package list for test environment. While adding package to list, if python is included, then ignore adding python to list for packages that test for valid python module/package imports.

I've manually tested this using miniconda3 for a build of a package that requires python 2.

### CODE REVIEW

**Comment by yannpaul** [comment]
FYI I'm newish to github, so please feel free to comment on how I can improve my pull requests. Also are there any suggestions/recommendations for testing?

**Comment by yannpaul** [comment]
I should note I also tested building package against python 3. It also ran smoothly.

**Comment by asmeurer** [comment]
As far as the git/GitHub stuff goes, it looks fine. The only suggestion I have there is to add the text "fixes #178" in the commit message so that it automatically closes the issue when merged. But it isn't that big of a deal if you don't do it.

File: conda_build/build.py [hunk 1]
```diff
@@ -458,15 +458,19 @@ def test(m, verbose=True, channel_urls=(), override_channels=False):
     rm_rf(config.test_prefix)
     specs = ['%s %s %s' % (m.name(), m.version(), m.build_id())]
 
-    if py_files:
+    # add packages listed in test/requires
+    specs_include_python = False
+    for spec in m.get_value('test/requires', []):
+        specs.append(spec)
+        if spec.startswith('python '):
```
Comment (asmeurer) [inline conda_build/build.py]: Won't this be wrong if the spec is just `python` (without a version)?

**Comment by asmeurer** [comment]
If you want to try adding a test the way we've been doing it is to have some test recipes in tests/test-recipes, which should not fail when built. We haven't been really great at testing things, though, so I won't fault you if you don't add a test here.

**Comment by yannpaul** [comment]
This might be wrong in regards to `python` without a version... I will test it out, as that's the standard requirement for pypi skeleton packages right? Is this issue at all related to #343 ?

**Comment by asmeurer** [comment]
I don't think it's related to that issue. 

Any package generated with skeleton isn't going to have python in the test requires, as test requirements already include the runtime requirements automatically, and every package generated by skeleton runtime requires python.

**Comment by yannpaul** [comment]
It seemed overkill to add build test cases if I'm testing one small component. I manually tested having run requirements (python 2.7*) different from CONDA_PY (python 3.4), with test requirements of various flavors (none, python, python 2.7*, python 3.4*). And everything seem to behave as expected.

File: conda_build/build.py [hunk 2]
```diff
@@ -458,15 +458,19 @@ def test(m, verbose=True, channel_urls=(), override_channels=False):
...
+
+    if py_files and not specs_include_python:
         # as the tests are run by python, we need to specify it
         specs += ['python %s*' % environ.get_py_ver()]
     if pl_files:
         # as the tests are run by perl, we need to specify it
         specs += ['perl %s*' % environ.get_perl_ver()]
-    # add packages listed in test/requires
-    for spec in m.get_value('test/requires', []):
-        specs.append(spec)
```
Comment (asmeurer) [inline conda_build/build.py]: Why did you delete these lines?

File: conda_build/build.py [hunk 3]
```diff
@@ -458,15 +458,19 @@ def test(m, verbose=True, channel_urls=(), override_channels=False):
     rm_rf(config.test_prefix)
     specs = ['%s %s %s' % (m.name(), m.version(), m.build_id())]
 
-    if py_files:
+    # add packages listed in test/requires
+    specs_include_python = False
+    for spec in m.get_value('test/requires', []):
+        specs.append(spec)
```
Comment (yannpaul) [inline conda_build/build.py]: @asmeurer They were just moved here, taken care of before the `py_files` check. I just think the diff is not picking up on the move.

Comment (asmeurer) [inline conda_build/build.py]: Oh I see now. Sorry.

**Comment by patricksnape** [comment]
This is a pretty horrible blocker for our automated builds at the moment. Would it be possible to get a bugfix release of conda build including this?

**Comment by asmeurer** [comment]
Does the workaround described at https://github.com/conda/conda-build/issues/178#issuecomment-52694641 work for you?
